src/protocol/reveal_additive_binary.rs

```diff
@@ -0,0 +1,103 @@
+use crate::{
+    error::BoxError,
+    helpers::{fabric::Network, prss::SpaceIndex, Direction},
+    protocol::{context::ProtocolContext, RecordId, Step},
+};
+
+use serde::{Deserialize, Serialize};
+
+/// A message sent by each helper when they've revealed their own shares
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct RevealValue {
+    share: bool,
+}
+
+/// This implements reveal algorithm
+/// As this is an additive sharing, each helper has just one boolean share
+/// As such, reveal requires each helper to send their share both left and right
+/// Put another way, each `P_i` sends `\[a\]_i` to `P_i+1` and `P_i-1`
+/// and then reconstructs a from` \[a\]_i`, `\[a\]_i+i` and `\[a\]i−1`.
+// Input: Each helpers know their own secret shares
+// Output: At the end of the protocol, all 3 helpers know a revealed (or opened) secret
+#[derive(Debug)]
+pub struct RevealAdditiveBinary {}
+
+impl RevealAdditiveBinary {
+    #[allow(dead_code)]
+    pub async fn execute<S: Step + SpaceIndex, N: Network<S>>(
+        ctx: &ProtocolContext<'_, S, N>,
+        step: S,
+        record_id: RecordId,
+        input: bool,
+    ) -> Result<bool, BoxError> {
+        let mut channel = ctx.gateway.get_channel(step);
+
+        channel
```

Comment:
We definitely should. I was just being lazy =). I'll fix it.

Reply:
No can do.
```
error[E0499]: cannot borrow `channel` as mutable more than once at a time
  --> src/protocol/reveal_additive_binary.rs:44:33
   |
37 |           let send_left_future = channel
   |  ________________________________-
38 | |             .send(
39 | |                 channel.identity().peer(Direction::Left),
40 | |                 record_id,
41 | |                 RevealValue { share: input },
42 | |             );
   | |_____________- first mutable borrow occurs here
43 |
44 |           let send_right_future = channel
   |  _________________________________^
45 | |             .send(
46 | |                 channel.identity().peer(Direction::Right),
47 | |                 record_id,
48 | |                 RevealValue { share: input },
49 | |             );
   | |_____________^ second mutable borrow occurs here
50 |
51 |           try_join(send_left_future, send_right_future).await?;
   |                    ---------------- first borrow later used here
```

help... is this something I can work around? Or is this a limitation in the infra we will need to address in a latter diff?

Reply:
OK, 
So I tried to create two different channels, so that I could send in parallel. It does not work. The subsequent "receive" never finishes. It just gets stuck forever.

I'm giving up on this... I hope someone else can help me make this parallel in a follow-up diff.